### PR TITLE
Dialog: normalize remaining ds4/ds6 structural differences

### PR DESCRIPTION
**Pull request by agliga**
## Description
Cleaned up dialog for ds4 and refactored some common code smells

## Context
* Added dialog-mini which has support for dialog-mini. Also added docs for it as well as animations
   * IAN NOTE: this may become part of lightbox module in a follow up PR
* Removed drawer ds4 code. It should be wrapped in the normal drawer code
* Removed DS4 code for header. The header was white in DS4, but it should be all the same color. There is also no centered header anymore, although we can fix that as well as the color with a bit of css changes and variables
* Removed support for headers outside of `dialog__header`
* Fixed animations to basically be more explicit removing any code smells. Changed animations to not rely on the position but rather they have the right names (ie `dialog__window--slide-right`)
* Removed `dialog__body`, it should be `dialog__main` now.

__Note__ Fixed that header overrides... I had copied some dialog code into the dialog-mini file which was causing that header issue that I found.

## References
#1167 
#1066

## Screenshots
Dialog mini
Screen Shot 2020-07-08 at 12 22 14 PM

Subpage
Screen Shot 2020-07-08 at 12 36 38 PM

DS4 Before:
Screen Shot 2020-07-08 at 12 38 20 PM

DS4 after:
Screen Shot 2020-07-08 at 12 38 46 PM

### CODE REVIEW

**Comment by ianmcburnie** [comment]
I was kind of thinking that mini dialog would be part of lightbox. Something like "lightbox lightbox--mini", although not totally decided yet. Can we revert this change and keep it till #1082 when we have a chance to think about the split-up as a whole?

**Comment by agliga** [comment]
> I was kind of thinking that mini dialog would be part of lightbox. Something like "lightbox lightbox--mini", although not totally decided yet. Can we revert this part of the PR and keep it till #1082 when we have a chance to think about the split-up as a whole?

Can we leave it as is for now and then fold that into the lightbox once we decide it? I think it will be much easier than having to revert that part.

**Comment by ianmcburnie** [comment]
Yes, that's fine. This can be work in progress.

File: dist/dialog-mini/ds4/dialog-mini.css [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+  left: 0;
+  -webkit-overflow-scrolling: touch;
+  overflow-y: auto;
+  position: fixed;
+  right: 0;
+  top: 0;
+  will-change: background-color;
+  z-index: 100000;
+}
+.dialog-mini--no-mask[role="dialog"] {
```
Comment (ianmcburnie) [inline dist/dialog-mini/ds4/dialog-mini.css:18]: Do we still need this no mask modifier? Is it a leftover from the really, really old markup we had? Or still used?

Comment (ianmcburnie) [reply dist/dialog-mini/ds4/dialog-mini.css:18]: Oops. comment was meant to be on the dialog.less file, not mini, but same applies.

**Comment by ianmcburnie** [comment]
Nitpick, but have a sweep over the docs too for anything ds4/ds6 specific:

Screen Shot 2020-07-08 at 7 17 10 PM

**Comment by agliga** [comment]
> Nitpick, but have a sweep over the docs too for anything ds4/ds6 specific:
> 
> Screen Shot 2020-07-08 at 7 17 10 PM

Done. I left the test/storybook intact because those are DS4 patterns that are no longer in DS6

File: dist/mixins/pagination/base/pagination-mixins.less [hunk 2]
```diff
@@ -1,14 +0,0 @@
-.pagination-layout-base() {
```
Comment (ianmcburnie) [inline dist/mixins/pagination/base/pagination-mixins.less:1]: I see there's a few unrelated mixin files deleted. Just checking this is intentional...

Comment (agliga) [reply dist/mixins/pagination/base/pagination-mixins.less:1]: I think those src files were deleted but when I reran the build, those got removed then. 
I don't see any src files but only the dist ones. Should those be there even if there is not src version of them?

Comment (ianmcburnie) [reply dist/mixins/pagination/base/pagination-mixins.less:1]: Ah, ok. Sounds good. Thanks. Will sign off.
